lang/rust/rust.py

```diff
@@ -0,0 +1,216 @@
+        actions.auto_insert(' = ')
+
+    def code_operator_subtraction():
+        actions.auto_insert(' - ')
+
+    def code_operator_subtraction_assignment():
+        actions.auto_insert(' -= ')
+
+    def code_operator_addition():
+        actions.auto_insert(' + ')
+
+    def code_operator_addition_assignment():
+        actions.auto_insert(' += ')
+
+    def code_operator_multiplication():
+        actions.auto_insert(' * ')
+
+    def code_operator_multiplication_assignment():
+        actions.auto_insert(' *= ')
+
+    def code_operator_exponent():
+        actions.auto_insert('.pow()');
+        actions.edit.left();
+
+    def code_operator_division():
+        actions.auto_insert(' / ')
+
+    def code_operator_division_assignment():
+        actions.auto_insert(' /= ')
+
+    def code_operator_modulo():
+        actions.auto_insert(' % ')
+
+    def code_operator_modulo_assignment():
+        actions.auto_insert(' %= ')
+
+    def code_operator_equal():
+        actions.auto_insert(' == ')
+
+    def code_operator_not_equal():
+        actions.auto_insert(' != ')
+
+    def code_operator_greater_than():
+        actions.auto_insert(' > ')
+
+    def code_operator_greater_than_or_equal_to():
+        actions.auto_insert(' >= ')
+
+    def code_operator_less_than():
+        actions.auto_insert(' < ')
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(' <= ')
+
+    def code_operator_and():
+        actions.auto_insert(' && ')
+
+    def code_operator_or():
+        actions.auto_insert(' || ')
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(' & ')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(' | ')
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(' ^ ')
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(' << ')
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(' <<= ')
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(' >> ')
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(' >>= ')
+
+    def code_import():
+        actions.auto_insert('use ')
+
+    def code_comment():
+        actions.auto_insert('// ')
+
+    def code_self():
+        actions.auto_insert('self')
+
+    def code_operator_object_accessor():
+        actions.auto_insert('.')
+
+    def code_insert_true():
+        actions.auto_insert('true')
+
+    def code_insert_false():
+        actions.auto_insert('false')
+
+    def code_state_if():
+        actions.insert('if {}')
+        for i in range(3):
+            actions.edit.left()
+
+    def code_state_else_if():
+        actions.insert('else if {}')
+        for i in range(3):
+            actions.edit.left()
+
+    def code_state_return():
+        actions.insert("return ")
+
+    def code_insert_function(text: str, selection: str):
+        actions.user.paste(f'fn {text}({selection or ""})')
```

Comment:
This action is used by `code_functions_gui` to insert function calls. I don't think you want `funk print line` to insert `fn println!()`, for example.

Reply:
Yeah, I will need to fix that, thank you!